src/irc.nim

```diff
@@ -571,9 +583,19 @@ proc connect*(irc: AsyncIrc) {.async.} =
   await irc.sock.connect(irc.address, irc.port)
   irc.status = SockConnected
 
-  if irc.serverPass != "": await irc.send("PASS " & irc.serverPass, true)
-  await irc.send("NICK " & irc.nick, true)
-  await irc.send("USER $1 * 0 :$2" % [irc.user, irc.realname], true)
+  if irc.nickservPass != "": 
+    await irc.send("CAP LS")
+    await irc.send("NICK " & irc.nick, true)
+    await irc.send("USER $1 * 0 :$2" % [irc.user, irc.realname], true)
+    await irc.send("CAP REQ :multi-prefix sasl")
+    await irc.send("AUTHENTICATE PLAIN")
+    let encodedpass = encode(char(0) &  irc.nick & char(0) & irc.nickservPass)
+    await irc.send("AUTHENTICATE " & encodedpass)
+    await irc.send("CAP END")
+  else:
+    if irc.serverPass != "": await irc.send("PASS " & irc.serverPass, true)
+    await irc.send("NICK " & irc.nick, true)
+    await irc.send("USER $1 * 0 :$2" % [irc.user, irc.realname], true)
```

Comment:
Would be nice to refactor this duplication, we should be able to with `{.multisync.}`. If you have some time would appreciate another PR for that :)